src/inference/graph_inference.cpp

```diff
@@ -920,60 +933,94 @@ class GraphNodeAssignment : public Nice2Assignment {
           factors_candidates.push_back(factors[j]);
         }
       }
+
+      // Go over all factors containing the labels in the "given" set, and create
+      // the initial permutation. It includes all labels in a factor that are not in the "given" set.
+      Factor giv_labels_copy = giv_labels;
```

Comment:
The less surprising choice is to:
1) make a copy, destroy the copy
You do:
2) make a copy, destroy the original, revert the original.